aws/rust-runtime/aws-sigv4/src/sign/v4a.rs

```diff
@@ -64,14 +59,14 @@ pub fn generate_signing_key(access_key: &str, secret_access_key: &str) -> impl A
         let tag = ring::hmac::sign(&key, &buf);
         let tag = &tag.as_ref()[0..32];
 
-        let k0 = BigInt::from_bytes_be(num_bigint::Sign::Plus, tag);
+        let k0 = U256::from_be_bytes(tag.try_into().expect("convert to [u8; 32]"));
 
         // It would be more secure for this to be a constant time comparison, but because this
         // is for client usage, that's not as big a deal.
-        if k0 <= *BIG_N_MINUS_2 {
-            let pk = k0 + BigInt::from(1i32);
-            let d = Zeroizing::new(pk.to_bytes_be().1);
-            break SigningKey::from_bytes(&d).unwrap();
+        if k0 <= BIG_N_MINUS_2 {
+            let pk = k0.checked_add(&U256::ONE).unwrap_or(U256::ZERO);
```

Comment:
This seems like it might be a change in behavior. What was the behavior for the `+` operator previously?

Reply:
It's not a change in behavior. `k0 + 1` is computed when `k0 <= BIG_N_MINUS_2`, and `BIG_N_MINUS_2 < U256::MAX`. I can change the `unwrap_or` to `expect` too.

Reply:
I think an `expect` with the reasoning above would be clearer.